apps/internal/base/internal/storage/storage_test.go

```diff
@@ -951,3 +932,52 @@ func TestWrite(t *testing.T) {
 		t.Fatalf("TestWrite(app metadata): -want/+got\n%s", diff)
 	}
 }
+
+func TestRemoveAccount(t *testing.T) {
+	accessTokenCacheItem := NewAccessToken(
+		"hid", "env", "realm", "cid",
+		time.Now(),
+		time.Now().Add(1000*time.Second),
+		time.Now(),
+		"openid profile", "secret",
+	)
+	testIDToken := NewIDToken("hid", "env", "realm", "cid", "secret")
+	testAppMeta := NewAppMetaData("fid", "cid", "env")
+	testRefreshToken := accesstokens.NewRefreshToken("hid", "env", "cid", "secret", "fid")
+	testAccount := shared.NewAccount("hid", "env", "realm", "lid", accAuth, "username")
+
+	contract := &Contract{
+		RefreshTokens: map[string]accesstokens.RefreshToken{
+			testRefreshToken.Key(): testRefreshToken,
+		},
+		Accounts: map[string]shared.Account{
+			testAccount.Key(): testAccount,
+		},
+		AppMetaData: map[string]AppMetaData{
+			testAppMeta.Key(): testAppMeta,
+		},
+		IDTokens: map[string]IDToken{
+			testIDToken.Key(): testIDToken,
+		},
+		AccessTokens: map[string]AccessToken{
+			accessTokenCacheItem.Key(): accessTokenCacheItem,
+		},
+	}
+	authParameters := authority.AuthParams{
+		HomeaccountID: "hid",
+		AuthorityInfo: authority.Info{},
+		ClientID:      "cid",
+		Scopes:        []string{"openid", "profile"},
+	}
+	discResp := authority.InstanceDiscoveryResponse{
+		TenantDiscoveryEndpoint: "tenant",
+		Metadata:                []authority.InstanceDiscoveryMetadata{},
+	}
+	responder := &fakeDiscoveryResponser{err: false, ret: discResp}
+	manager := newForTest(responder)
+	manager.update(contract)
+	err := manager.RemoveAccount(testAccount, authParameters.ClientID)
```

Comment:
should test that all item are removed as expected, and items that should not be removed are not removed

Reply:
Done https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/215/commits/16a8e58fbe4c78cd9c807cb83eae2fa35cb07ccb